src/main/java/controller/BaseballManagement.java

```diff
@@ -0,0 +1,119 @@
+        while(ReGameFlag == 1) {
+            run();
+        }
+    }
+
+    public void run() {
+        setPlayer(createRandomNumber());
+        while(GameFlag) {
+            inputUserBaseballNumber();
+            strikeOrBallOrOutProcessing();
+            endGame();
+        }
+        ReGameFlag = Integer.parseInt(inputView.inputMenu());
+        GameFlag = true;
+    }
+
+    public void setPlayer(List<Integer> RandomNumberList){
+        player = new Player(RandomNumberList);
+    }
+
+    private List<Integer> createRandomNumber(){
+        List<Integer> randomNumberList = new ArrayList<Integer>();
+        new Random()
+                .ints(1,10)
+                .limit(LIMITNUMBER)
+                .boxed()
+                .distinct()
+                .forEach(num->randomNumberList.add(num));
+
+        return checkRandomNumberList(randomNumberList);
+    }
+
+    private List<Integer> checkRandomNumberList(List<Integer> randomNumberList) {
+        return (randomNumberList.size() == LIMITNUMBER) ? randomNumberList : createRandomNumber();
+    }
+
+    public void inputUserBaseballNumber() {
+        userBaseballNumber = new ArrayList<>();
+        convertToIntegerAdd(inputView.inputBaseballNumber());
+    }
+
+    private void convertToIntegerAdd(List<String> stringList) {
+        for (int i = 0; i < stringList.size(); i++) {
+            userBaseballNumber.add(Integer.parseInt(stringList.get(i)));
+        }
+    }
+
+    private List<Integer> makeCompareList() {
+        List<Integer> compareList = new ArrayList<>();
+        userBaseballNumber.stream()
+                .forEach(userNum->player.getBaseballNum().stream()
+                                                        .forEach(comNum->compare(userNum,comNum, compareList)));
+        return compareList;
+    }
+    private long checkBall(List<Integer> compareList) {
+        long ballCount = IntStream.range(1,compareList.size())
+                .filter(i -> i%4 > 0 && compareList.get(i) == 1)
+                .count();
+        if(ballCount != 0) {
+            outputView.ballPrint(ballCount);
+        }
+        return ballCount;
+    }
+
+    private long checkStrike(List<Integer> compareList) {
+        strikeCount = IntStream.range(0,compareList.size())
+                .filter(i -> i%4 == 0 && compareList.get(i) == 1)
+                .count();
+        if(strikeCount != 0){
+            outputView.strikePrint(strikeCount);
+        }
+
+        GameFlag = (strikeCount == LIMITNUMBER) ? false  : true;
+
+        return strikeCount;
+    }
+
+    public void strikeOrBallOrOutProcessing() {
+        List<Integer> compareList = makeCompareList();
+        long outCheck = checkBall(compareList) + checkStrike(compareList);
+        if(outCheck == 0){
+            outputView.outPrint();
+        }
+    }
+
+    private void compare(Integer userNum, Integer comNum, List<Integer> compareList) {
+        compareList.add((userNum == comNum) ? 1 : 0);
```

Comment:
```suggestion
        compareList.add((userNum.equals(comNum)) ? 1 : 0);
```
Integer는 객체이기 때문에 동등성으로 비교해 주어야 합니다~
한번 동일성과 동등성에 대해 찾아보고 공부해보세요~ :)
(hascode()와 equals() 메소드에 대해서)

참고 : [https://anster.tistory.com/160](https://anster.tistory.com/160)